### PR TITLE
added a new option to config form of Custom Provider: now the "No valid One Time Link" message is configurable

### PR DESCRIPTION
**Pull request by aleevas**
**https://fivejars.atlassian.net/browse/YGRVY-28:**

## Steps to test:

- [ ] Log in as admin
- [ ] Go to the  Virtual YMCA Auth settings page (/admin/openy/virtual-ymca/gc-auth-settings)
- [ ] Please make sure that you have the **Custom** provider is enabled
- [ ] Go to the edit this provider page (/admin/openy/virtual-ymca/gc-auth-settings/provider/custom)
- [ ] In the  **Email verification** fieldset please find a new field **Message for One Time Link no valid**
YGRVY-28NewOPtion

- [ ] You can change the default massage
- [ ] Save the form
- [ ] Go as anonymous to the wrong One Time LInkfor ex.: /vy-user/verification/25/1611744498/fa9ab0e40af7fc745a0c62d88a23993a)
- [ ] You should see your own customized error message
YGRVY-28ErrorMessage

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [ ] This PR  provides updates via `hook_update_N` or other means.
  - [ ] No updates are necessary for this change.
- Front end fixes should be tested against all of the Open Y Themes.
  - [ ] Tested against Carnation
  - [ ] Tested against Lily
  - [ ] Tested against Rose
  - [ ] This change does not contain front-end fixes.
- [ ] I have flagged this PR "Needs Review" or pinged the VY devs/QA
  team in Slack

### CODE REVIEW

File: modules/openy_gc_auth/modules/openy_gc_auth_custom/config/install/openy_gc_auth.provider.custom.yml [hunk 1]
```diff
@@ -4,3 +4,4 @@ require_email_verification: 0
 email_verification_link_life_time: '14400'
 email_verification_text: 'Hello! <br> You’re just one step away from accessing your Virtual YMCA. Please open the link below to begin enjoying YMCA content made exclusively for members like you.'
 verification_message: 'We have sent a verification link to the email address you provided. Please open this link and activate your account. If you do not receive an email, please try again or contact us at XXX-XXX-XXXX to ensure we have the correct email on file for your membership.'
+one_time_link_no_valid_message: 'You have tried to use a one-time login link that has either been used or is no longer valid. Please request a new one using the form below.'
```
Comment (hamrant) [inline modules/openy_gc_auth/modules/openy_gc_auth_custom/config/install/openy_gc_auth.provider.custom.yml:7]: Let's rename to `one_time_link_invalid_message`

Comment (hamrant) [reply modules/openy_gc_auth/modules/openy_gc_auth_custom/config/install/openy_gc_auth.provider.custom.yml:7]: And please add an upgrade path for this param

File: modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Plugin/GCIdentityProvider/Custom.php [hunk 2]
```diff
@@ -155,6 +163,7 @@ public function submitConfigurationForm(array &$form, FormStateInterface $form_s
       $this->configuration['email_verification_link_life_time'] = $form_state->getValue('email_verification_link_life_time');
       $this->configuration['email_verification_text'] = !empty($form_state->getValue('email_verification_text')) ? $form_state->getValue('email_verification_text')['value'] : '';
       $this->configuration['verification_message'] = !empty($form_state->getValue('verification_message')) ? $form_state->getValue('verification_message')['value'] : '';
+      $this->configuration['one_time_link_no_valid_message'] = !empty($form_state->getValue('one_time_link_no_valid_message')) ? $form_state->getValue('one_time_link_no_valid_message'): '';
```
Comment (hamrant) [inline modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Plugin/GCIdentityProvider/Custom.php:166]: Please add `one_time_link_invalid_message` to `defaultConfiguration()`

**Review by hamrant — CHANGES_REQUESTED** [review]

**Comment by hamrant** [comment]
@aleevas please fix code sniffer error - http://ci.fivejars.com:8080/blue/organizations/jenkins/OPENY_GC_PR/detail/OPENY_GC_PR/2558/pipeline

File: modules/openy_gc_auth/modules/openy_gc_auth_custom/config/install/openy_gc_auth.provider.custom.yml (hunk 1 above)
Comment (aleevas) [reply modules/openy_gc_auth/modules/openy_gc_auth_custom/config/install/openy_gc_auth.provider.custom.yml]: @hamrant thanks for review!
Fixed. Please re-check it.

File: modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Plugin/GCIdentityProvider/Custom.php (hunk 2 above)
Comment (aleevas) [reply modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Plugin/GCIdentityProvider/Custom.php]: @hamrant 
Fixed. Please re-check it.

File: modules/openy_gc_auth/modules/openy_gc_auth_custom/openy_gc_auth_custom.install [hunk 3]
```diff
@@ -136,3 +136,14 @@ function openy_gc_auth_custom_update_8007(&$sandbox) {
     ->set('require_email_verification', 0)
     ->save();
 }
+
+/**
+ * Add value for one_time_link_invalid_message setting.
+ */
+function openy_gc_auth_custom_update_8008(&$sandbox) {
+  $message = (string) t('You have tried to use a one-time login link that has either been used or is no longer valid. Please request a new one using the form below.');
```
Comment (hamrant) [inline modules/openy_gc_auth/modules/openy_gc_auth_custom/openy_gc_auth_custom.install:144]: No need to use translation here.

**Review by hamrant — CHANGES_REQUESTED** [review]

File: modules/openy_gc_auth/modules/openy_gc_auth_custom/config/install/openy_gc_auth.provider.custom.yml [hunk 4]
```diff
@@ -3,4 +3,5 @@ enable_email_verification: 1
 require_email_verification: 0
 email_verification_link_life_time: '14400'
 email_verification_text: 'Hello! <br> You’re just one step away from accessing your Virtual YMCA. Please open the link below to begin enjoying YMCA content made exclusively for members like you.'
-verification_message: 'We have sent a verification link to the email address you provided. Please open this link and activate your account. If you do not receive an email, please try again or contact us at XXX-XXX-XXXX to ensure we have the correct email on file for your membership.'
+  verification_message: 'We have sent a verification link to the email address you provided. Please open this link and activate your account. If you do not receive an email, please try again or contact us at XXX-XXX-XXXX to ensure we have the correct email on file for your membership.'
```
Comment (anpolimus) [inline modules/openy_gc_auth/modules/openy_gc_auth_custom/config/install/openy_gc_auth.provider.custom.yml:6]:
```suggestion
verification_message: 'We have sent a verification link to the email address you provided. Please open this link and activate your account. If you do not receive an email, please try again or contact us at XXX-XXX-XXXX to ensure we have the correct email on file for your membership.'
```

File: modules/openy_gc_auth/modules/openy_gc_auth_custom/openy_gc_auth_custom.install (hunk 3 above)
Comment (anpolimus) [inline modules/openy_gc_auth/modules/openy_gc_auth_custom/openy_gc_auth_custom.install:144]:
```suggestion
  $message = 'You have tried to use a one-time login link that has either been used or is no longer valid. Please request a new one using the form below.';
```

**Comment by anpolimus** [comment]
Ready to test.

**Comment by AnastasiiaPys** [comment]
tested, looks good for me.

**Pull request merged**